doc/api/fs.md

```diff
@@ -4342,6 +4362,17 @@ Following successful read, the `Promise` is resolved with an object with a
 `bytesRead` property specifying the number of bytes read, and a `buffer`
 property that is a reference to the passed in `buffer` argument.
 
+#### `filehandle.read(options)`
+<!-- YAML
+added: REPLACEME
+-->
+* `options` {Object}
+  * `buffer` {Buffer|Uint8Array}
```

Comment:
I assume the defaults here are the same as the `fs.read(fd, options, callback)` version? If so, they should be documented here too.

Reply:
Also, I might be missing it but is the Promise version tested?

Reply:
i don't think you are missing anything.  i think i forgot to add a test for the promise based call

Reply:
> I assume the defaults here are the same as the fs.read(fd, options, callback) version? If so, they should be documented here too.

Right,  added them and a test for the promise in the latest 2 commits

Also, I noticed that the buffer types for the `filehandle.read` function are not the same as the regular `fs.read` function buffer types. That doesn't seem correct to me